### PR TITLE
Shannon/ad fade

### PR DESCRIPTION
**Pull request by smarie33**
#### What does this PR do & why?
- create fade in / fade out for rail ads

#### How can this be manually tested?
- navigate to any post page, like: http://localhost:8080/janes-embeds-test-post-1625982
- watch the ad in the right rail for 30 seconds to watch it fade out for 1 second and fade in for 1 second
- to speed this up, modify the rail component to rotate ads less that 30 seconds
  - navigate to components > Rail > index.js > line 27

#### Any questions or concerns about this change?
- i feel like 2 seconds is too long, so i made fades 1 second each, (which together is 2 seconds). let me know if you want to 2 seconds

#### Link to story card and other documentation
https://app.asana.com/0/543787180217047/759556625512872

#### Have you remembered to...?

* [ ] add a reviewer
* [ ] run appropriate tests
* [ ] tag anyone who might want to see this
* [ ] update any relevant wiki documentation

### CODE REVIEW

File: components/Ad/index.scss [hunk 1]
```diff
@@ -2,3 +2,14 @@
   display: flex;
   justify-content: center;
 }
+
+.Rail {
```
Comment (andrewb) [inline components/Ad/index.scss]: We don't want to limit fading to rail ads. That's our only use now, but there's nothing stopping us from setting a refresh interval for any ad, in which case we'll want it to fade as well.

File: components/Ad/index.js [hunk 2]
```diff
@@ -1,30 +1,46 @@
...
 import './index.scss';
 
 class Ad extends Component {
   constructor(props) {
     super(props);
     this.setRefreshTimeout = this.setRefreshTimeout.bind(this);
+    this.slotDelayFade = this.slotDelayFade.bind(this);
     this.slotRefreshTimeout = '';
+    this.slotDelayFadeTimer = '';
+    this.state = { showAd: false };
```
Comment (andrewb) [inline components/Ad/index.js]: I think we should reverse this logic so it's `fadeAd` instead. At the moment ads that don't have a `refreshInterval` will have a `showAd` value of `false` which could get confusing.

File: components/Ad/index.scss [hunk 3]
```diff
@@ -2,3 +2,14 @@
...
+  .Ad {
+    opacity: 0;
+    transition: 1s;
+    &--show {
+      opacity: 1;
+      transition: 1s;
```
Comment (andrewb) [inline components/Ad/index.scss]: I think we should play it safe and only transition on `opacity`.

File: components/Ad/index.js (hunk 2 above)
Comment (andrewb) [reply components/Ad/index.js]: That way `.Ad` can have a default opacity of `1` as well.

File: components/Ad/index.js [hunk 4]
```diff
@@ -1,18 +1,27 @@
...
 import { Bling as GPT } from 'react-gpt';
+import classnames from 'classnames';
 import { getSizeProp } from './helpers';
 import './index.scss';
 
 class Ad extends Component {
   constructor(props) {
     super(props);
+
+    if (this.props.refreshInterval) {
```
Comment (andrewb) [inline components/Ad/index.js]: Nitpick, you could simplify to `this.state = { hideAd: !!this.props.refreshInterval }`

File: components/Ad/index.js [hunk 5]
```diff
@@ -1,18 +1,27 @@
...
+      this.state = { hideAd: true };
+    } else {
+      this.state = { hideAd: false };
+    }
     this.setRefreshTimeout = this.setRefreshTimeout.bind(this);
+    this.slotDelayFade = this.slotDelayFade.bind(this);
```
Comment (andrewb) [inline components/Ad/index.js]: Nitpick, would a better name for this be `this.setFadeTimeout`?

File: components/Ad/index.js [hunk 6]
```diff
@@ -1,18 +1,27 @@
...
     this.handleOnBlur = this.handleOnBlur.bind(this);
     this.handleOnFocus = this.handleOnFocus.bind(this);
     this.startTimers = this.startTimers.bind(this);
     this.clearTimers = this.clearTimers.bind(this);
     this.slotRefreshTimeout = undefined;
+    this.slotDelayFadeTimer = undefined;
```
Comment (andrewb) [inline components/Ad/index.js]: Nit, I'd call this `slotFadeTimeout` so it's consistent with `slotRefreshTimeout`.

File: components/Ad/index.js (hunk 4 above)
Comment (smarie33) [reply components/Ad/index.js]: 2079942

**Pull request merged**